variables.tf

```diff
@@ -55,18 +55,25 @@ variable "location" {
   description = "The default location used for resources in this Landing Zone."
 }
 
+variable "terraform_version" {
+  type        = string
+  description = "Terraform version to install."
+}
+
 variable "vnet_config" {
   type = object({
     dns_server    = list(string)
     address_space = string
     subnets       = map(string)
+    nsg           = bool
   })
   description = <<-DOC
   ```
   If you want to provide a virtual network, please provide the following values: 
     dns_server: DNS Servers that will be used in the network.
     address_space: Address space of the virtual network in CIDR notation.
     subnets: Subnets that will be created in the virtual network. Use 'Usecase' as the key and the address prefix as the value in CIDR notation.    
+    nsg: Create NSG for all the subnets.

```

Comment:
This variable does not control the creation of NSGs just the association. Effectively this means the same but I would either change the name/description or make the variable control the creation instead.